### PR TITLE
ESLint config external

### PR DESCRIPTION
**Pull request by webpro**
* Fix reusability of .prettierrc file finder in eslint config (3862e375)
* Fix dependencies for eslint packages (e8e3a5d5)

### CODE REVIEW

File: packages/tools/eslint-config/profile/lib.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+  const filePath = path.join(dir, filename);
+  if (fs.existsSync(filePath)) return filePath;
+  const parentDir = path.dirname(dir);
+  if (parentDir === dir) return;
+  return findUp(filename, parentDir);
+};
+
 const prettierOptions = JSON.parse(
-  fs.readFileSync(path.resolve(__dirname, '../../../../.prettierrc'), 'utf8'),
+  fs.readFileSync(findUp('.prettierrc'), 'utf8'),
```
Comment (alber70g) [inline packages/tools/eslint-config/profile/lib.js:13]: How does this work when this package is a dependency of a package that has a `.prettierrc` and resides in the `node_modules`?
Maybe we should consider going up to the higest `node_modules` dir first.

Something like (pseudocode)
```
const nodeModulesPath = findUpHighest(`node_modules`)
const firstNonNodeModulesPrettierrcPath = findUp(from: nodeModulesPath, `.prettierrc`)
```

Comment (alber70g) [reply packages/tools/eslint-config/profile/lib.js:13]: This should cover it

`findUp('.prettierrc', process.cwd().split('node_modules')[0])`

Comment (webpro) [reply packages/tools/eslint-config/profile/lib.js:13]: This surely isn't flawless, but I simply tested the following scenarios:

- Run within this monorepo from a package dir, e.g. `rushx lint` or `rush lint --only ...`
- Run within this monorepo from root with `rush lint`
- Run from root in another repo

Guess that's alright for now?

Comment (alber70g) [reply packages/tools/eslint-config/profile/lib.js:13]: I'd say so. But it's a small change to make it work everywhere, no? 

```js
findUp('.prettierrc', __dirname.split('node_modules')[0])
```

**Review by alber70g — APPROVED** [review]

File: packages/tools/eslint-config/profile/lib.js (hunk 1 above)
Comment (webpro) [reply packages/tools/eslint-config/profile/lib.js:13]: Discussed offline that the current solution is good/intentional.

**Pull request merged**